### PR TITLE
UI-40 add confirm modal

### PR DESCRIPTION
**Pull request by lmiller1990**
Added `<Modal>` which you can use to render anything as a modal. Also added `<ConfirmModal>` which you use like this:

```
<Modal>
  <ConfirmModal />
</Modal>
```

Seems the most flexible.

image

### CODE REVIEW

File: src/Icons/TrashFill.tsx [hunk 1]
```diff
@@ -1,8 +1,13 @@
 import React from 'react'
 
-const TrashFill: React.FC = () => {
+interface IProps {
```
Comment (lmiller1990) [inline src/Icons/TrashFill.tsx:3]: should update all icons to take props at some point by extending a base class or something

File: src/ConfirmModal/ConfirmModal.tsx [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  entityName?: string
+  onCancel: () => any
+  onConfirm: () => any
+}
+
+const capitalizeString = (str: string) => {
+  return str.replace(/^\w/, c => c.toUpperCase())
+}
+
+const ConfirmModal: React.FC<IProps> = (props) => {
```
Comment (MichaelDorkhom) [inline src/ConfirmModal/ConfirmModal.tsx:19]: Do we want to make this modal delete specific? Or should we make it more generic? (i.e. `type = 'critical' | 'warning' | 'info'` and the ability to pass the warning/info message as a prop)

File: src/Modal/index.scss [hunk 3]
```diff
@@ -0,0 +1,20 @@
+@import '../style/colors.scss';
+
+.overlay {
+  position: fixed;
+  top: 0;
+  left: 0;
+  background-color: rgba(0, 0, 0, 0.5);
```
Comment (MichaelDorkhom) [inline src/Modal/index.scss:7]: Can we use a generic color here

Comment (lmiller1990) [reply src/Modal/index.scss:7]: There is no generic color with an alpha channel, is there?

File: src/ConfirmModal/ConfirmModal.tsx (hunk 2 above)
Comment (lmiller1990) [reply src/ConfirmModal/ConfirmModal.tsx:19]: This is an interesting one. So far the mock-ups for Vamp don't have any other varieties. Let's add those when we know how they will look, and have a better idea of what the inputs/outputs are.

**Pull request merged**